drone-impl/src/main/java/org/jboss/arquillian/drone/impl/DeploymentDronePointsRegistry.java

```diff
@@ -0,0 +1,67 @@
+package org.jboss.arquillian.drone.impl;
+
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.drone.spi.DronePoint;
+
+/**
+ * This registry keeps all {@link DronePoint}s with the {@link DronePoint.Lifecycle.DEPLOYMENT} scope and the reference
+ * to the instance of the test class the DronePoint is declared in
+ *
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class DeploymentDronePointsRegistry {
```

Comment:
well... no, there is no registry per Drone point lifecycle. 
Do you mean that you would welcome a registry for all the Drone points? There is something similar - DroneContext - it keeps all the Drone points, however, some additional changes would be needed (related to the inner implementation) and this class is available there in the SPI. So, I didn't want to change the SPI because of these changes and didn't want to make the information available through the SPI. The reason why I created a new separate registry is because this will be needed only in one case - when a deployment-scoped drone point is used - so mostly the registry won't be needed and won't be created. Not sure if it is good approach or not - that's why I'm asking for the review :-).